### PR TITLE
image viewer

### PR DESCRIPTION
**Pull request by jsyrtsov**

### CODE REVIEW

File: TheMovieDatabase/ViewControllers/Favorites/View/FavoritesViewController.swift [hunk 1]
```diff
@@ -58,6 +59,23 @@ final class FavoritesViewController: UIViewController {
...
+    private func refreshData() {
+        let service = MoviesService()
+        service.loadFavoriteMovies(accountId: accountId) { (movies) in
+            guard let movies = movies else {
+                return
+            }
+            self.movies = movies
+            self.tableView.reloadData()
+            self.refreshControl.endRefreshing()
+        }
```
Comment (ismetanin) [inline TheMovieDatabase/ViewControllers/Favorites/View/FavoritesViewController.swift]: Looks like a duplicate of `loadFavoriteMovies`
What do you think about unification?

File: TheMovieDatabase/ViewControllers/Feed/View/FeedViewController.swift [hunk 2]
```diff
@@ -45,6 +46,14 @@ final class FeedViewController: UIViewController {
...
         segmentedControl?.addTarget(self, action: #selector(FeedViewController.indexChanged(_:)), for: .valueChanged)
 
+        refreshControl.attributedTitle = NSAttributedString(string: "Pull to refresh")
+        refreshControl.addTarget(self, action: #selector(refreshData), for: .valueChanged)
+        tableView.refreshControl = refreshControl
+    }
+
+    @objc
+    private func refreshData() {
+        loadMovies()
```
Comment (ismetanin) [inline TheMovieDatabase/ViewControllers/Feed/View/FeedViewController.swift]: You can directly call `loadMovies`
I mean you don't need to create `refreshData` method, you can add `@objc` to your `loadMovies` and pass it as a selector

File: TheMovieDatabase/ViewControllers/FullPictureScroll/View/FullPictureScrollViewController.swift [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+//  FullPictureScrollViewController.swift
+//  TheMovieDatabase
+//
+//  Created by Evgeny Syrtsov on 5/1/20.
+//  Copyright © 2020 Evgeny Syrtsov. All rights reserved.
+//
+
+import UIKit
+
+final class FullPictureScrollViewController: UIViewController {
```
Comment (ismetanin) [inline TheMovieDatabase/ViewControllers/FullPictureScroll/View/FullPictureScrollViewController.swift]: I don't like `FullPicture` name :(
What does it mean? Full picture? How does a picture maybe full or not full? 

Also, I haven't heard anything about `ScrollViewController` can you shed some light on this?

Comment (ismetanin) [reply TheMovieDatabase/ViewControllers/FullPictureScroll/View/FullPictureScrollViewController.swift]: My suggestions: `ImageViewerViewController`, `GalleryViewController` (first one is better for me)

File: TheMovieDatabase/ViewControllers/FullPictureScroll/View/FullPictureScrollViewController.swift [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+            state = .normal
+        }
+    }
+}
+
+// MARK: - UIScrollViewDelegate
+
+extension FullPictureScrollViewController: UIScrollViewDelegate {
+    func scrollViewDidEndDecelerating(_ scrollView: UIScrollView) {
+        if scrollView == self.scrollView {
```
Comment (ismetanin) [inline TheMovieDatabase/ViewControllers/FullPictureScroll/View/FullPictureScrollViewController.swift]: Now you have only one scroll view so you can skip this condition

File: TheMovieDatabase/ViewControllers/FullPictureScroll/View/FullPictureScrollViewController.swift (hunk 3 above)
Comment (ismetanin) [reply TheMovieDatabase/ViewControllers/FullPictureScroll/View/FullPictureScrollViewController.swift]: Note: it's not okay to leak implementation details through a name. There you leaked that you implemented your controller with a scroll view. Maybe it's okay when you have couple implementations, but it's not your case

File: TheMovieDatabase/ViewControllers/Person/View/PersonViewController.swift [hunk 5]
```diff
@@ -180,7 +180,9 @@ extension PersonViewController: UICollectionViewDataSource {
 extension PersonViewController: UICollectionViewDelegate {
     func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
         if collectionView == imagesCollectionView {
-            output?.showFullPicture(picturePath: personImages[indexPath.row].filePath)
+            let images = personImages.map { $0.filePath }
+            output?.showFullPictureScroll(images: images, currentImage: indexPath.row)
+            //output?.showFullPicture(picturePath: personImages[indexPath.row].filePath)
```
Comment (ismetanin) [inline TheMovieDatabase/ViewControllers/Person/View/PersonViewController.swift]: 😕

**Review by ismetanin — CHANGES_REQUESTED** [review]

File: TheMovieDatabase/ViewControllers/Feed/View/FeedViewController.swift (hunk 2 above)
Comment (jsyrtsov) [reply TheMovieDatabase/ViewControllers/Feed/View/FeedViewController.swift]: good idea

File: TheMovieDatabase/ViewControllers/Favorites/View/FavoritesViewController.swift (hunk 1 above)
Comment (jsyrtsov) [reply TheMovieDatabase/ViewControllers/Favorites/View/FavoritesViewController.swift]: nice ideaa

File: TheMovieDatabase/ViewControllers/Person/View/PersonViewController.swift (hunk 5 above)
Comment (jsyrtsov) [reply TheMovieDatabase/ViewControllers/Person/View/PersonViewController.swift]: sorry, forgot about this

**Review by ismetanin — APPROVED** [review]

**Pull request merged**